schema/pod.go

```diff
@@ -116,6 +116,11 @@ func (al AppList) assertValid() error {
 			return fmt.Errorf(`duplicate apps of name %q`, a.Name)
 		}
 		seen[a.Name] = true
+		if a.App != nil {
```

Comment:
But look at the diff. Marshalling wasn't being called on it before you added it. I'm saying, why do you need to add this stanza at all?